terraform/ec2/linux/main.tf

```diff
@@ -142,6 +144,56 @@ resource "null_resource" "integration_test" {
   ]
 }
 
+// Only run on canary test
```

Comment:
Would it be best to separate the canary workflow though? Since I believe the canary is for confirm if the latest agent is able to run. Therefore, does not need some resources such as `efs` and it won't stop here since we have container use case too

Reply:
EFS is needed for consistent disk metrics across runs. If disk is required metric to track, it is good to have efs. As for adding an ecs canary, I would say extend the ecs terraform code. Adding a new terraform package for canary means copying lots of code and for most things changing in both places on change. There is very little difference between canary and a regular integration test.

Reply:
I understand your point but somehow, I would disagree with that. 
* The whole point of canary is to verify if the agent is running normally with the latest version, its not confirming the metrics are sending. The metrics are sending is part of the integration test/feature testing/plugin testing 
* I would disagree regarding extending the ecs terraform code. Even though I agree the major use case currently are EC2, we are moving to container (which is a trending based on Kevin). so there is a possible that we will only use ECS/EKS (be it on EC2 or Fargate) to do it canary and its covers EC2
*  Somehow I'm not convinced though regarding `difference  between canary and a regular integration test`.

Reply:
@jefchien is setting up alarms to the canary. That is the validation.